modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/WhisperEngine.java

```diff
@@ -145,7 +172,26 @@ public File generateSubtitlesFile(File mediaFile, File preparedOutputFile, Strin
       IoSupport.closeQuietly(process);
     }
 
+    // Detect language if not set
+    if (language.isBlank()) {
+      JSONParser jsonParser = new JSONParser();
+      try {
+        FileReader reader = new FileReader((preparedOutputFile.getParent() + "/" + mediaFile.getName() + ".json"));
```

Comment:
Don't know how language detection works with whisper, but this does not work for me:
```
	2023-04-05 17:16:28,319 | ERROR | (AbstractJobProducer$JobRunner:343) - Error handling operation 'speechtotext':
org.opencastproject.speechtotext.api.SpeechToTextServiceException: Error while generating subtitle from http://localhost:8080/assets/assets/e838d659-d8c1-47a4-9868-4f42061b4d90/ed9a9e83-a295-41b7-88e9-efd922d1b06d/2/WasistChaos_flash9.mp4
	at org.opencastproject.speechtotext.impl.SpeechToTextServiceImpl.process(SpeechToTextServiceImpl.java:168) ~[?:?]
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:313) [!/:?]
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:272) [!/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: org.opencastproject.speechtotext.api.SpeechToTextEngineException: java.io.FileNotFoundException: /home/arnewilken/IdeaProjects/opencast13-2/build/opencast-dist-develop-13-SNAPSHOT/data/opencast/workspace/collection/subtitles/WasistChaos_flash9.mp4.json (Datei oder Verzeichnis nicht gefunden)
	at org.opencastproject.speechtotext.impl.engine.WhisperEngine.generateSubtitlesFile(WhisperEngine.java:187) ~[?:?]
	at org.opencastproject.speechtotext.impl.SpeechToTextServiceImpl.process(SpeechToTextServiceImpl.java:157) ~[?:?]
	... 6 more
Caused by: java.io.FileNotFoundException: /home/arnewilken/IdeaProjects/opencast13-2/build/opencast-dist-develop-13-SNAPSHOT/data/opencast/workspace/collection/subtitles/WasistChaos_flash9.mp4.json (Datei oder Verzeichnis nicht gefunden)
	at java.io.FileInputStream.open0(Native Method) ~[?:?]
	at java.io.FileInputStream.open(FileInputStream.java:219) ~[?:?]
	at java.io.FileInputStream.<init>(FileInputStream.java:157) ~[?:?]
	at java.io.FileInputStream.<init>(FileInputStream.java:112) ~[?:?]
	at java.io.FileReader.<init>(FileReader.java:60) ~[?:?]
	at org.opencastproject.speechtotext.impl.engine.WhisperEngine.generateSubtitlesFile(WhisperEngine.java:180) ~[?:?]
	at org.opencastproject.speechtotext.impl.SpeechToTextServiceImpl.process(SpeechToTextServiceImpl.java:157) ~[?:?]
	... 6 more
```